lambdas/tests/unit/services/token-request-validator.test.ts

```diff
@@ -0,0 +1,168 @@
+import { AccessTokenRequestValidator } from "../../../src/services/token-request-validator";
+import { JwtVerifierFactory } from "../../../src/common/security/jwt-verifier";
+import { SessionItem } from "../../../src/types/session-item";
+describe("token-request-validator.ts", () => {
+    let accessTokenRequestValidator: AccessTokenRequestValidator;
+    const mockJwtVerifierFactory = jest.mocked(JwtVerifierFactory);
+    const code = "test";
+    const redirect_uri = "http://abc123.com";
+    const client_assertion = "test";
+    const client_assertion_type = "urn:ietf:params:oauth:client-assertion-type:jwt-bearer";
+    const grant_type = "authorization_code";
+
+    beforeEach(() => {
+        jest.resetAllMocks();
+        accessTokenRequestValidator = new AccessTokenRequestValidator(mockJwtVerifierFactory.prototype);
+    });
+
+    describe("validatePayload", () => {
+        it("should throw when the client_assertion_type is not valid", function () {
+            const tokenRequestBody = `code=${code}&redirect_uri=${redirect_uri}&client_assertion=${client_assertion}&client_assertion_type=test&grant_type=${grant_type}`;
+            let requestPayload;
+            try {
```

Comment:
I generally prefer using `expect(() => accessTokenRequestValidator.validatePayload(tokenRequestBody)).toThrow("Invalid grant_type parameter")` because it avoids relying on the catch statement for your expectation to be run - if it doesn't error then the statement is missed completely

Reply:
The problem is also avoided with those expect assertions(2) statement because then the test knows it needs to hit both

Reply:
Good spot, will update